### PR TITLE
detect false positive in "not for people" rules

### LINKED ISSUE #1075: detect if word applies to person, organization or thing
- employee

- team

- \.\.\.

<p></p>

review posthog <span data-preserve-white-space></span>

expand list via LLM

### PR DESCRIPTION
**Pull request by lsmith77**
to handle cases like 'flexible employees' vs. 'flexible company'

fixes https://github.com/witty-works/nlp_api/issues/1075

requires https://github.com/witty-works/rule-editor/pull/95

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -5707,6 +5746,55 @@ async def rule_check(
...
+
+                    skip = False
+                else:
+                    # Assume the first noun chunk applies
+                    token_chunk = chunks[0]
+
+            if token_chunk is not None:
+                chunk_token_index = token_chunk.start
+                while chunk_token_index < token_chunk.end:
+                    # TODO fill list of words
```
Comment (lsmith77) [inline app/main.py]: @hellasol I guess we will need to add a flag to german/english nouns. we should be able to fill this using an LLM.

File: tests/test_demo_wordings_english/test_internal_email/output.json [hunk 2]
```diff
@@ -180,28 +180,34 @@
...
             },
             "gravity": 2.0,
-            "label": "Ability + Physicality: Medical state",
+            "label": "Social Motive: Agentic language",
             "proficiency_level": "unconscious_bias",
             "start": 111,
-            "subcategory": "medical_state",
+            "subcategory": "agentic",
             "text": "challenged",
-            "text_id": "challenged"
```
Comment (lsmith77) [inline tests/test_demo_wordings_english/test_internal_email/output.json]: check

File: tests/test_general_cases/test_api_explanation_context/output.json [hunk 3]
```diff
@@ -3,37 +3,22 @@
...
+                "text": "Stärkt die Zusammenarbeit und den Sinn für Gemeinschaft",
+                "url": "https://www.witty.works/de/kategorien/soziale-motivation/kommunale-sprache?reducedView=true"
             },
-            "gravity": 3.0,
-            "label": "Kulturelle Diversität: Worthülsen",
-            "proficiency_level": "unconscious_bias",
-            "start": 8,
-            "subcategory": "hollow_advanced",
-            "text": "hoch motiviert",
-            "text_id": "hoch motiviert"
```
Comment (lsmith77) [inline tests/test_general_cases/test_api_explanation_context/output.json]: check

File: tests/test_not_for_people/test_english/input.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "text": "Employees are flexible, so are companies. You are flexible. We are looking for flexible employees for our flexible company."
```
Comment (lsmith77) [inline tests/test_not_for_people/test_english/input.json:2]: missing one

File: tests/test_not_for_people/test_german/input.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "text": "Mitarbeitende sind flexibel, Firmen auch. Du bist flexibel. Wir suchen flexible Mitarbeitende für unsere flexible Firma."
```
Comment (lsmith77) [inline tests/test_not_for_people/test_german/input.json:2]: missing one

File: tests/test_spacy_model/text_with_space/output.json [hunk 6]
```diff
@@ -1,39 +1,5 @@
...
-                },
-                {
-                    "text": "Main"
-                },
-                {
-                    "text": "Primary"
-                }
-            ],
-            "category": "social-motive",
-            "context": "Are you Dominant",
```
Comment (lsmith77) [inline tests/test_spacy_model/text_with_space/output.json]: check

File: tests/test_spacy_model/text_without_space/output.json [hunk 7]
```diff
@@ -1,39 +1,5 @@
...
-                },
-                {
-                    "text": "Main"
-                },
-                {
-                    "text": "Primary"
-                }
-            ],
-            "category": "social-motive",
-            "context": "Are you Dominant",
```
Comment (lsmith77) [inline tests/test_spacy_model/text_without_space/output.json]: check

File: tests/test_witty_addons/test_api_hr_addon_off/output.json [hunk 8]
```diff
@@ -5,46 +5,6 @@
...
-                "text": "Häufig verwendete Floskeln werden oft als nichtssagend empfunden.",
-                "url": "https://www.witty.works/de/kategorien/kulturelle-diversität/worthülsen?reducedView=true"
-            },
-            "gravity": 3.0,
-            "label": "Kulturelle Diversität: Worthülsen",
-            "proficiency_level": "unconscious_bias",
-            "start": 37,
-            "subcategory": "hollow_advanced",
-            "text": "Arbeitskraft",
-            "text_id": "Arbeitskraft"
```
Comment (lsmith77) [inline tests/test_witty_addons/test_api_hr_addon_off/output.json]: check

File: tests/test_witty_addons/test_api_hr_addon_on/output.json [hunk 9]
```diff
@@ -60,46 +60,6 @@
...
-                "text": "Häufig verwendete Floskeln werden oft als nichtssagend empfunden.",
-                "url": "https://www.witty.works/de/kategorien/kulturelle-diversität/worthülsen?reducedView=true"
-            },
-            "gravity": 3.0,
-            "label": "Kulturelle Diversität: Worthülsen",
-            "proficiency_level": "unconscious_bias",
-            "start": 37,
-            "subcategory": "hollow_advanced",
-            "text": "Arbeitskraft",
-            "text_id": "Arbeitskraft"
```
Comment (lsmith77) [inline tests/test_witty_addons/test_api_hr_addon_on/output.json]: check

File: tests/test_witty_addons/test_api_hr_addon_unset/output.json [hunk 10]
```diff
@@ -60,46 +60,6 @@
...
-                "text": "Häufig verwendete Floskeln werden oft als nichtssagend empfunden.",
-                "url": "https://www.witty.works/de/kategorien/kulturelle-diversität/worthülsen?reducedView=true"
-            },
-            "gravity": 3.0,
-            "label": "Kulturelle Diversität: Worthülsen",
-            "proficiency_level": "unconscious_bias",
-            "start": 37,
-            "subcategory": "hollow_advanced",
-            "text": "Arbeitskraft",
-            "text_id": "Arbeitskraft"
```
Comment (lsmith77) [inline tests/test_witty_addons/test_api_hr_addon_unset/output.json]: check

File: tests/test_witty_free/test_api/output.json [hunk 11]
```diff
@@ -20,46 +20,6 @@
...
-                "text": "Häufig verwendete Floskeln werden oft als nichtssagend empfunden.",
-                "url": "https://www.witty.works/de/kategorien/kulturelle-diversität/worthülsen?reducedView=true"
-            },
-            "gravity": 3.0,
-            "label": "Kulturelle Diversität: Worthülsen",
-            "proficiency_level": "unconscious_bias",
-            "start": 11,
-            "subcategory": "hollow_advanced",
-            "text": "eine Arbeitskraft",
-            "text_id": "Arbeitskraft"
```
Comment (lsmith77) [inline tests/test_witty_free/test_api/output.json]: check

**Review by hellasol — APPROVED** [review]

**Pull request merged**